themes/bootstrap3/js/check_item_statuses.js

```diff
@@ -88,147 +89,143 @@ VuFind.register('itemStatuses', function ItemStatuses() {
+        });
+      },
+      success: itemStatusAjaxSuccess,
+      failure: itemStatusAjaxFailure,
+      delay,
+    });
+  }
 
   //store the handlers in a "hash" obj
   var checkItemHandlers = {
-    'ils': ItemStatusHandler,
-    'overdrive': OdItemStatusHandler,
+    ils: makeItemStatusQueue(),
+    overdrive: makeItemStatusQueue({ url: "/Overdrive/getStatus" }),
   };
 
   function checkItemStatus(el) {
-    var $item = $(el);
-    if ($item.hasClass('js-item-pending') || $item.hasClass('js-item-done')) {
+    const hiddenIdEl = el.querySelector(".hiddenId");
+
+    if (
+      hiddenIdEl === null ||
+      el.classList.contains("js-item-pending") ||
+      el.classList.contains("js-item-done")
+    ) {
       return;
     }
-    if ($item.find('.hiddenId').length === 0) {
-      return;
+
+    let handlerName = "ils";
+    if (el.dataset.handlerName) {
+      handlerName = el.dataset.handlerName;
+    } else {
+      const handlerNameEl = el.querySelector(".handler-name");
+
+      if (handlerNameEl !== null) {
+        handlerName = handlerNameEl.value;
+      }
     }
-    var id = $item.find('.hiddenId').val();
-    var handlerName = 'ils';
-    if ($item.data("handler-name")) {
-      handlerName = $item.data("handler-name");
-    } else if ($item.find('.handler-name').length > 0) {
-      handlerName = $item.find('.handler-name').val();
+
+    // update element to reflect lookup
+    el.classList.add("js-item-pending");
+    el.classList.remove("hidden");
+    el.querySelector(".callnumAndLocation").classList.remove("hidden");
+    el.querySelector(".callnumAndLocation .ajax-availability").classList.remove("hidden");
+    const statusEl = el.querySelector(".status");
+    if (statusEl) {
+      statusEl.classList.remove("hidden");
     }
 
-    //queue the element into the handler
-    checkItemHandlers[handlerName].itemQueueAjax(id, $item);
+    //queue the element into the queue
+    checkItemHandlers[handlerName].add({ el, id: hiddenIdEl.value });
   }
 
-  function checkItemStatuses(_container) {
-    var container = typeof _container === 'undefined'
-      ? document.body
-      : _container;
-
-    var ajaxItems = $(container).find('.ajaxItem');
-    for (var i = 0; i < ajaxItems.length; i++) {
-      checkItemStatus($(ajaxItems[i]));
-    }
+  function checkAllItemStatuses(container = document) {
+    container.querySelectorAll(".ajaxItem").forEach(checkItemStatus);
   }
-  function init(_container) {
-    var container = typeof _container === 'undefined'
-      ? document.body
-      : _container;
+
+  function init($container = document) {
+    const container = $container instanceof Node ? $container : $container[0];
 
     if (VuFind.isPrinting()) {
-      checkItemStatuses(container);
-      return;
+      checkAllItemStatuses(container);
+    } else {
+      VuFind.observerManager.createIntersectionObserver(
+        'itemStatuses',
+        checkItemStatus,
+        (container).querySelectorAll('.ajaxItem')
```

Comment:
I don't think the parentheses around `(container)` serve any purpose here. Worth simplifying?

Reply:
I'm gonna test this out soon as I get my cup of coffee and check if there is something to be done.

One thing I'm not sure about is the removal of check: checkItemStatuses, checkRecord: checkItemStatus in check_item_statuses.js return part. If somebody is using them directly, it might come as a suprise that those are removed. Should they be marked as deprecated first and be removed later?